infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationServer.java

```diff
@@ -331,6 +331,12 @@ private void sendLeadershipLoss(@Nonnull RequestMsg request,
     }
 
     public void setLeadership(boolean leader) {
+        // Leadership change can come from sinkClientRouter and also from interClusterServerNode.
```

Comment:
LogReplicationServer is used by SINK always, irrespective of whether is the connection starter/receiver.
1. when connections starter, it will receive `setLeadership(false)` while the router is shutting down.
2. setLeadership(false) is also called when the cluster looses leadership in the discoveryService.

The above 2 are async.

Have updated the comment to say "Leadership change can come from sinkClientRouter when SINK is the connection starter and the router is shutting down, and also from discoveryService when connection endpoint and is no longer the leader."